src/spikeinterface/preprocessing/deepinterpolation/train.py

```diff
@@ -0,0 +1,307 @@
+from __future__ import annotations
+import json
+import os
+import warnings
+from pathlib import Path
+from typing import Optional
+
+from concurrent.futures import ProcessPoolExecutor
+import multiprocessing as mp
+
+from .tf_utils import import_tf
+
+# from .generators import define_recording_generator_class
+
+from ...core import BaseRecording
+
+
+global train_func
+
+
+def train_deepinterpolation(
+    recordings: BaseRecording | list[BaseRecording],
+    model_folder: str | Path,
+    model_name: str,
+    desired_shape: tuple[int, int],
+    train_start_s: Optional[float] = None,
```

Comment:
To what extent using seconds is helpful here? I am always worried about edge conditions created by wrapping code.

Reply:
I think that using second here is much more convenient. E.g. I'd rather use 600 to indicate 10 minutes, than 18000000 (easy to forget, add a 0 here)